substratevm/src/com.oracle.svm.core.posix/src/com/oracle/svm/core/posix/PosixJavaNetSubstitutions.java

```diff
@@ -119,84 +419,1339 @@ protected int peekData(DatagramPacket p) throws IOException {
+            //  1537          CHECK_NULL_RETURN(addr, NULL);
+
+            //  1539          setInetAddress_addr(env, addr, ntohl(in.s_addr));
+            JavaNetNetUtil.setInetAddress_addr(addr, NetinetIn.ntohl(inP.s_addr()));
+            //  1540          JNU_CHECK_EXCEPTION_RETURN(env, NULL);
+
+            //  1542          /*
+            //  1543           * For IP_MULTICAST_IF return InetAddress
+            //  1544           */
+            //  1545          if (opt == java_net_SocketOptions_IP_MULTICAST_IF) {
+            if (opt == SocketOptions.IP_MULTICAST_IF) {
+                //  1546              return addr;
+                return addr;
+            }
+
+            //  1549          /*
+            //  1550           * For IP_MULTICAST_IF2 we get the NetworkInterface for
+            //  1551           * this address and return it
+            //  1552           */
+            //  1553          if (ni_class == NULL) {
+            //  1554              jclass c = (*env)->FindClass(env, "java/net/NetworkInterface");
+            //  1555              CHECK_NULL_RETURN(c, NULL);
+            //  1556              ni_ctrID = (*env)->GetMethodID(env, c, "<init>", "()V");
+            //  1557              CHECK_NULL_RETURN(ni_ctrID, NULL);
+            //  1558              ni_indexID = (*env)->GetFieldID(env, c, "index", "I");
+            //  1559              CHECK_NULL_RETURN(ni_indexID, NULL);
+            //  1560              ni_addrsID = (*env)->GetFieldID(env, c, "addrs",
+            //  1561                                              "[Ljava/net/InetAddress;");
+            //  1562              CHECK_NULL_RETURN(ni_addrsID, NULL);
+            //  1563              ni_nameID = (*env)->GetFieldID(env, c,"name", "Ljava/lang/String;");
+            //  1564              CHECK_NULL_RETURN(ni_nameID, NULL);
+            //  1565              ni_class = (*env)->NewGlobalRef(env, c);
+            //  1566              CHECK_NULL_RETURN(ni_class, NULL);
+            //  1567          }
+            //  1568          ni = Java_java_net_NetworkInterface_getByInetAddress0(env, ni_class, addr);
+            ni = Target_java_net_NetworkInterface.getByInetAddress0(addr);
+            //  1569          if (ni) {
+            if (ni != null) {
+                //  1570              return ni;
+                return ni;
+            }
+
+            //  1573          /*
+            //  1574           * The address doesn't appear to be bound at any known
+            //  1575           * NetworkInterface. Therefore we construct a NetworkInterface
+            //  1576           * with this address.
+            //  1577           */
+            //  1578          ni = (*env)->NewObject(env, ni_class, ni_ctrID, 0);
+            ni = Util_java_net_NetworkInterface.newNetworkInterface();
+            //  1579          CHECK_NULL_RETURN(ni, NULL);
```

Comment:
You should add a comment saying that `Util_java_net_NetworkInterface.newNetworkInterface()` will never return null.

Similarly below after `new InetAddress[1]`.

Reply:
Nah, never mind.  No one else does that.